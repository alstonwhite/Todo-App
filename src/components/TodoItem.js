import React from "react";

const TodoItem = ({ item, onSelect, onDelete, onComplete }) => {
  if (item.status === "new") {
    return (
      <li>
        <button className="button task-button" onClick={() => onSelect(item)}>
          {item.title}
        </button>
        <button
          className="button button-small button-delete"
          onClick={() => onDelete(item)}
        >
          X
        </button>
      </li>
    );
  } else {
    return (
      <li>
        <button className="button task-button" onClick={() => onSelect(item)}>
          {item.title}
        </button>
        <button className="button button-small button-complete">âœ”</button>
      </li>
    );
  }
};

export default TodoItem;
