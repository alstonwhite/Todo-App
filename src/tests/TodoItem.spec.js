import React from "react";
import renderer from "react-test-renderer";
import ReactTestUtils from 'react-dom/test-utils';

import TodoItem from "../components/TodoItem";

it("Renders a list correctly (new status)", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "new",
        body: "Add notes..."
      }
    ];
    const selectTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoItem
          item={item}
          onSelect={selectTask}
          onDelete={deleteTask}
          onComplete={completeTask}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
  });

  it("Renders a list correctly (completed status)", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const selectTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoItem
          item={item}
          onSelect={selectTask}
          onDelete={deleteTask}
          onComplete={completeTask}
        />
      );
      expect(tree.toJSON()).toMatchSnapshot();
  });

  it("Select task button triggers function on button click", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const selectTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoItem
          item={item}
          onSelect={selectTask}
          onDelete={deleteTask}
          onComplete={completeTask}
        />
      );
    let button = tree.root.findByProps({className:'button task-button'});
    ReactTestUtils.Simulate.click(button);
    expect(selectTask).toHaveBeenCalled();
  });

  it("Delete task button triggers function on button click", () => {
    const item = [
      {
        _id: 1,
        title: "New task...",
        status: "completed",
        body: "Add notes..."
      }
    ];
    const selectTask = jest.fn();
    const deleteTask = jest.fn();
    const completeTask = jest.fn();
    const tree = renderer
      .create(
        <TodoItem
          item={item}
          onSelect={selectTask}
          onDelete={deleteTask}
          onComplete={completeTask}
        />
      );
    let button = tree.root.findByProps({className:'button button-small button-delete'});
    ReactTestUtils.Simulate.click(button);
    expect(deleteTask).toHaveBeenCalled();
  });

// Renders: Item with new status, item with completed status
// Task button: Calls proper fn, updates activeItem (App level)
// Delete button: Calls proper fn, removes item from list (arr length decreases) (App level), removes correct item (by ._id)