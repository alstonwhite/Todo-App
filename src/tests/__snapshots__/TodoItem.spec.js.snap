// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a list correctly (completed status) 1`] = `
<li>
  <button
    className="button task-button"
    onClick={[Function]}
  />
  <button
    className="button button-small button-complete"
  >
    ✔
  </button>
</li>
`;

exports[`Renders a list correctly (new status) 1`] = `
<li>
  <button
    className="button task-button"
    onClick={[Function]}
  />
  <button
    className="button button-small button-complete"
  >
    ✔
  </button>
</li>
`;

exports[`renders correctly 1`] = `
<li>
  <button
    className="button task-button"
    onClick={[Function]}
  />
  <button
    className="button button-small button-complete"
  >
    ✔
  </button>
</li>
`;

exports[`renders correctly 2`] = `
<li>
  <button
    className="button task-button"
    onClick={[Function]}
  />
  <button
    className="button button-small button-complete"
  >
    ✔
  </button>
</li>
`;
